### DISCUSSION TITLE
Configuring LTSP clients to mount image from different NFS servers

### DISCUSSION THREAD

**Discussion by vsuojanen** [ltsp/ltsp#609] [General] [answered]
I have a LTSP network with ~20  users booting from one LTSP server (chrootless, proxydhcp). 
I configure small client specific settings in ltsp.conf  and test the  changes in  cloned development server and afterwards configure the settings changes in production.

When I  build a new client image I setup the apps and changes first to a cloned LTSP server. The new LTSP server is taken to production with different IP address from the old one. Before starting the new server in the client network I shutdown  the proxdhcp/dhcp from the new server with 
`ltsp dnsmasq --proxy-dhcp=0 --real-dhcp=0`

When old server is running proxydhcp and new server running without proxydhcp/dhcp I disable the proxydhcp from the old server and start it on the new server 
`ltsp dnsmasq ` 

Everything in the new server with different ip address is prepared (the client image, tftp and nfsroot and ipxe). This normally works well when I'm confident with both the new server and old server as I need to take care only the other server is running as proxydhcp server
Now I have a four users to whom I want to start using a new image from new server. I want to run the new server couple days just for specific client macs and then one by one move the clients to mount image from the new server. If i use my "old" method I can not prevent other clients booting from the new server too early. 

I'm sorry if i'm generic here I have just no issue yet. I can test all the cases and methods of course here but before I go testing has anyone already played with the ltsp.ipxe configuring 
1. different mac sections 
2. and new image {img} section 
3. that client also get kernel parameters so that it mounts the nfsroot from different [srv}? 

Does it break the LTSP client boot if you configure in iPXE to mount the nfsroot from different nfs server?
I was to go fast here and avoid running separate image names, copying /srv/tftp/ltsp/x86_64 binaries from the new server and renaming them for the different client image
Any other recommendations for this kind of setup?

I have not taken serious actions yet, just thinking this path like
1. manually new client MAC specific sections in iPXE (ltsp.ipxe)
2. manually or with DEFAULT_IMAGE=the-image-you-want (ltsp.conf) the new image section to ltsp.ipxe
3. possible different client kernel params (ltsp.conf) to ltsp.ipxe ? i don't know if this is needed but I think the kernel, ltsp.img and initrd should exist in the boot server right?

**Comment by alkisg** [comment, marked as answer]
From the [dnsmasq man page](https://thekelleys.org.uk/dnsmasq/docs/dnsmasq-man.html) read the `--dhcp-host=` paragraph.
That way you can have the primary server ignore a few clients, i.e. `dhcp-ignore=tag:test-clients`
and the secondary server answer only to them, i.e. `dhcp-ignore=tag:!test-clients`.
That way you'll be able to test with the secondary server without any changes in networking, without running `ltsp dnsmasq` etc; everything in one subnet.
You can also use the SSH_SERVER that I mentioned in ltsp.conf to point them to use the primary server for authentication and /home.

When you're done, you just change a line in dnsmasq and you can boot all the clients from the primary server again.

I don't see any need for KERNEL_PARAMETERS or DEFAULT_IMAGE for this use case.

**Reply by vsuojanen** [reply]
Primary server:
dhcp-host=00:16:3e:88:10:b2,00:16:3e:3a:4d:bf,00:15:5d:6d:8f:0c,set:test-clients
dhcp-ignore=tag:test-clients

Secondary server:
dhcp-host=00:16:3e:88:10:b2,00:16:3e:3a:4d:bf,00:15:5d:6d:8f:0c,set:test-clients
dhcp-ignore=tag:!test-clients

**Comment by vsuojanen** [comment]
Thanks, 
testing configuring /etc/dnsmasq.d/ltsp-dnsmasq.conf with dhcp-host tag process and setting dhcp-ignore.

First configuring

Primary server:
`dhcp-host=00:16:3e:88:10:b2,00:16:3e:3a:4d:bf,00:15:5d:6d:8f:0c,set:test-clients`
`dhcp-ignore=tag:test-clients`

Secondary server:
`dhcp-ignore=tag:!test-clients`

This did not boot the test-clients, the pxe requests were ignored for test-clients also on the secondary server:

> Nov 17 13:56:37 oriented-titmouse dnsmasq-dhcp[1314]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored
> Nov 17 13:56:45 oriented-titmouse dnsmasq-dhcp[1314]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored

After  I configured the dhcp-host tag also on the secondary server:

`dhcp-host=00:16:3e:88:10:b2,00:16:3e:3a:4d:bf,00:15:5d:6d:8f:0c,set:test-clients`
`dhcp-ignore=tag:!test-clients`

Now the test-clients were ignored on the primary server as planned but this time they boot on the secondary server:

> Nov 17 14:03:02 oriented-titmouse dnsmasq-dhcp[1314]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored
> Nov 17 14:03:09 oriented-titmouse dnsmasq-dhcp[1314]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored
> 
> Nov 17 14:03:02 oriented-titmouse dnsmasq-dhcp[509]: PXE(eth0) 00:15:5d:6d:8f:0c proxy
> Nov 17 14:03:06 oriented-titmouse dnsmasq-dhcp[509]: PXE(eth0) 00:15:5d:6d:8f:0c proxy
> Nov 17 14:03:07 oriented-titmouse dnsmasq-tftp[509]: error 8 User aborted the transfer received from ?
> Nov 17 14:03:07 oriented-titmouse dnsmasq-tftp[509]: failed sending /srv/tftp/ltsp/snponly.efi to ?
> Nov 17 14:03:07 oriented-titmouse dnsmasq-tftp[509]: sent /srv/tftp/ltsp/snponly.efi to ?
> Nov 17 14:03:09 oriented-titmouse dnsmasq-dhcp[509]: PXE(eth0) 00:15:5d:6d:8f:0c proxy
> Nov 17 14:03:09 oriented-titmouse dnsmasq-dhcp[509]: PXE(eth0) 00:15:5d:6d:8f:0c proxy
> Nov 17 14:03:13 oriented-titmouse dnsmasq-tftp[509]: sent /srv/tftp/ltsp/ltsp.ipxe to ?
> Nov 17 14:04:50 oriented-titmouse dnsmasq-tftp[509]: sent /srv/tftp/ltsp/x86_64/vmlinuz to ?
> Nov 17 14:04:51 oriented-titmouse dnsmasq-tftp[509]: sent /srv/tftp/ltsp/ltsp.img to ?

This was not so good example to read because I used the same hostname on both primary server and secondary server

So without dhcp-host tag also on the secondary server the dnsmasq dhcp-host tag trick doesn't work (confirmed)
Maybe I did not first understand the two dnsmasq proxydhcp setup

Primary server:

> Nov 17 14:23:11 oriented-titmouse dnsmasq-dhcp[1314]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored
> Nov 17 14:23:20 oriented-titmouse dnsmasq-dhcp[1314]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored
> Nov 17 14:23:24 oriented-titmouse dnsmasq-dhcp[1314]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored
> 

Secondary server:

> Nov 17 14:23:11 oriented-titmouse dnsmasq-dhcp[602]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored
> Nov 17 14:23:20 oriented-titmouse dnsmasq-dhcp[602]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored
> Nov 17 14:23:24 oriented-titmouse dnsmasq-dhcp[602]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored
> Nov 17 14:23:32 oriented-titmouse dnsmasq-dhcp[602]: PXE(eth0) 00:15:5d:6d:8f:0c proxy-ignored

**Comment by madagh-one** [comment]
Hello  Every body, and an Happy new Year 2022.

I am new here,  and sorry if my post is not in good place, and too sorry of my bad English !!

I am interest of the solution of LTSP, i want realy put it in some associations, i tried it with a simple installation to following the web site documentary of it on vvirtual machine ( virtual box), and it was working, 

But i had a lot off of not understanding for what i want do and some tutorial are not explicite or maybe ia m not understanding good too !

I want to know if i can had a different distribution to send it in light client from LTSP, if my ltsp is ubuntu server can i use like debian light client ? if yes how ? how i build the image file which place ......, i am not expert on linux i am learning every days by my own, 

How i can build a different users and different password of client light ? 

How i can send the different image distribution on specific light client ? 

How i can limit the the user of light client to modify some sittings like desktop image , an autorization of he want instal some software, ......

And yes i want try different things with LTSP, with virtualisation server or real server too.

Please, as i said i am not an expert of linux and that project are verry intersting todeploy it and some small structur, for perform it and secure it and more 

thank you for understanding me. 

i wish got a real help on it
